src/handlers/records-query.ts

```diff
@@ -210,4 +214,22 @@ export class RecordsQueryHandler implements MethodHandler {
   private static shouldProtocolAuthorizeQuery(recordsQuery: RecordsQuery): boolean {
     return recordsQuery.authorSignaturePayload!.protocolRole !== undefined;
   }
+
+  /**
+   * Does the recordQuery filter explicitly for published records.
+   * Checks that there is either an explicit datePublished filter, or published is explicitly not set to false.
```

Comment:
Sanity:

> published is explicitly not set to false.
 
Is this different to explicitly set to true? What is the expected return value when `filter.published` is `undefined`? Assuming `true`, then the method name should be renamed to `filterIncludesPublishedRecords()`

Reply:
@thehenrytsai if `published` is undefined the filter for that property isn't evaluated at all.
That's today's behavior without this change. So a DWN owner would get both published/unpublished without it.

For a non-owner, we create multiple queries depending on the author/recipient/protocol auth and injecting additional filtered properties to the individual queries.

I wonder how I could make this intent clearer.

Reply:
Same as the other comment. Can we rename to `filterIncludesPublishedRecords()`.